TextTemplatingFileGeneratorCore/TextTemplatingFileGeneratorCore.cs

```diff
@@ -99,28 +100,154 @@ private IEnumerable<string> ProcessReferences(string[] references, string inputF
             ThreadHelper.ThrowIfNotOnUIThread();
 
             DTE dte = (DTE)GetService(typeof(DTE));
+            // Solution
+            string solutionName = string.Empty;
+            string solutionExt = string.Empty;
+            string solutionFileName = string.Empty;
+            string solutionDir = string.Empty;
+            string solutionPath = string.Empty;
+            // Project
+            string projectName = string.Empty;
+            string projectExt = string.Empty;
+            string projectFileName = string.Empty;
+            string projectDir = string.Empty;
+            string projectPath = string.Empty;
+            // Target
+            string targetName = string.Empty;
+            string targetExt = string.Empty;
+            string targetFileName = string.Empty;
+            string targetDir = string.Empty;
+            string targetPath = string.Empty;
+            // Other
+            string configurationName = string.Empty;
+            string outDir = string.Empty;
+            string platformName = string.Empty;
+            string devEnvDir = string.Empty;
+
+            if( dte != null )
+            {
+                if( dte.FullName != null )
+                {
+                    devEnvDir = Path.GetDirectoryName(dte.FullName);
+                    if( devEnvDir != null && !devEnvDir.EndsWith(@"\") )
+                    {
+                        devEnvDir += @"\";
```

Comment:
Only appends a slash if it doesn't have one to prevent accidentical double slash